src/app/pages/interest/interest.js

```diff
@@ -1,178 +1,189 @@
+    return (
+        <React.Fragment>
+            <HiddenFields leadSource="Interest Form" />
+            <input type="hidden" name="Role__c" value={selectedRole} />
+            <ContactInfo validatorRef={validatorRef} />
+        </React.Fragment>
+    );
+}
+
+function firstSearchArgument() {
+    return decodeURIComponent(window.location.search.substr(1))
+        .replace(/&.*/, '');
+}
+
+function HowDidYouHear() {
+    const options = [
+        {value: 'Web search', label: 'Web search'},
+        {value: 'Colleague', label: 'Colleague'},
+        {value: 'Conference', label: 'Conference'},
+        {value: 'Email', label: 'Email'},
+        {value: 'Facebook', label: 'Facebook'},
+        {value: 'Twitter', label: 'Twitter'},
+        {value: 'Webinar', label: 'Webinar'},
+        {value: 'Partner organization', label: 'Partner organization'}
+    ];
+
+    return (
+        <FormCheckboxgroup name="How_did_you_Hear__c"
+            longLabel="How did you hear about OpenStax?"
+            instructions="Select all that apply (optional)."
+            options={options}
+        />
+    );
+}
+
+function BookSelectorPage({selectedBooksRef, bookBeingSubmitted}) {
+    const preselectedTitle = firstSearchArgument();
+    const [selectedBooks, toggleBook] = useSelectedBooks();
+
+    selectedBooksRef.current = selectedBooks;
+    return (
+        <div className="page-2">
+            <BookSelector prompt="Which textbook(s) are you currently using?"
+                required
+                selectedBooks={selectedBooks}
+                preselectedTitle={preselectedTitle}
+                toggleBook={toggleBook}
+            />
+            <FormInput
+                longLabel="How many students do you teach each semester?"
+                inputProps={{
+                    name: 'Number_of_Students__c',
+                    type: 'number',
+                    min: '1',
+                    max: '999',
+                    required: true
+                }}
+            />
+            <HowDidYouHear />
+            {
+                bookBeingSubmitted &&
+                    <React.Fragment>
+                        <input type="hidden" name="Subject__c" value={bookBeingSubmitted.value} />
+                        <input type="hidden" name="First__c" value={bookBeingSubmitted === selectedBooks[0]} />
+                    </React.Fragment>
+            }
+        </div>
+    );
+}
+
+function FacultyForm({selectedRole, onPageChange}) {
+    const salesforce = useDataFromPromise(salesforcePromise, {});
+    const contactValidatorRef = useRef();
+    const selectedBooksRef = useRef();
+    const [currentBook, setCurrentBook] = useState();
+    const formRef = useRef();
+
+    function validatePage(page) {
+        const validateContactInfo = contactValidatorRef.current;
+
+        if (page === 1 && !validateContactInfo()) {
+            return false;
+        }
+        return true;
     }
```

Comment:
a quibble, but you could: `return Boolean(page !== 1 && validateContactInfo())`

Reply:
with ||, yes.